### DISCUSSION TITLE
Roda behind reverse proxy with context path

### DISCUSSION THREAD

**Discussion by gugrim** [keeps/roda#3292] [Q&A] [answered]
Hi!

I'm trying to set up a test/demo installation of Roda on a docker host running several other services. We use a reverse proxy to direct different context paths such as '/portainer' to different containers. I'd like to run Roda on a the context path '/roda'.
This doesn't work since the Roda web application assumes its resources is available on the root.

Does Roda have an option like Portainer's --base-url to specify that it is accessed below a context path?

/Gunnar

**Comment by luis100** [comment]
We use Spring, please check this options for setting context path.

https://www.baeldung.com/spring-boot-context-path

**Comment by gugrim** [comment]
Thanks for the tip but I can't get it to work. I modified docker-compose.yaml to add the environment variable for the context path. Added the one for Spring Boot 1 also:

```
    roda:
    :
    environment:
      :
      - SERVER_SERVLET_CONTEXT_PATH=/roda
      - SERVER_CONTEXT_PATH=/roda
```
No effect. Roda is still served on the root context. Is the environment setting overridden by some other configuration or code?

**Comment by luis100** [comment, marked as answer]
Actually, I see that Spring Boot is only on the development branch, to be released in RODA 6 (or if you use the :development docker tag).

In RODA 5, we use Tomcat, and its being deployed in the ROOT context. To change it, you might need to change the Dockerfile, or inject a configuration to tomcat to redefine the context path of the ROOT web application.
